### PR TITLE
Added workflow URLScan with Microsoft Teams

### PR DESCRIPTION
**Pull request by tylerslijboom-r7**
## Requirements

Added workflow URLScan with Microsoft Teams

#### Workflow Builder

workflow1
workflow2

#### Job

job1

#### Artifact

artifact1

### CODE REVIEW

File: workflows/URLScan_with_urlscanio_from_Microsoft_Teams/URLScan_with_Microsoft_Teams.icon [hunk 1]
```diff
@@ -0,0 +1,1728 @@
+{
+  "kom": {
+    "komandVersion": "1.49.2-0-gafb52",
+    "komFileVersion": "2.0.0",
+    "exportedAt": "2020-04-23T16:48:11.225694111Z",
+    "workflowVersions": [
+      {
+        "name": "URLScan with urlscan.io from Microsoft Teams",
```
Comment (jschipp-r7) [inline workflows/URLScan_with_urlscanio_from_Microsoft_Teams/URLScan_with_Microsoft_Teams.icon:8]:
```suggestion
        "name": "Enrich URL with urlscan.io from Microsoft Teams",
```

File: workflows/URLScan_with_urlscanio_from_Microsoft_Teams/help.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
+# Description
+
+This workflow looks up a URL given via Slack command in urlscan.io and reports information on the URL back to Microsoft Teams.
+
+Sample Teams Trigger Commands:
+
+`!urlscan https://bit.ly/39HoYMu`
```
Comment (jschipp-r7) [inline workflows/URLScan_with_urlscanio_from_Microsoft_Teams/help.md:7]:
```suggestion
`!enrich-url https://bit.ly/39HoYMu`
```

File: workflows/URLScan_with_urlscanio_from_Microsoft_Teams/help.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+# Key Features
+
+* Lookup URL from Microsoft Teams
```
Comment (jschipp-r7) [inline workflows/URLScan_with_urlscanio_from_Microsoft_Teams/help.md:11]:
```suggestion
* Enrich URL from Microsoft Teams
```

File: workflows/URLScan_with_urlscanio_from_Microsoft_Teams/workflow.spec.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
+extension: workflow
+products: ["insightconnect"]
+name: URLScan_with_urlscanio_from_Microsoft_Teams
+title: "URLScan with urlscan.io from Microsoft Teams"
```
Comment (jschipp-r7) [inline workflows/URLScan_with_urlscanio_from_Microsoft_Teams/workflow.spec.yaml:4]:
```suggestion
title: "Enrich URL with urlscan.io from Microsoft Teams"
```

File: workflows/URLScan_with_urlscanio_from_Microsoft_Teams/workflow.spec.yaml [hunk 5]
```diff
@@ -0,0 +1,31 @@
+extension: workflow
+products: ["insightconnect"]
+name: URLScan_with_urlscanio_from_Microsoft_Teams
```
Comment (jschipp-r7) [inline workflows/URLScan_with_urlscanio_from_Microsoft_Teams/workflow.spec.yaml:3]: Update directory name workflow file name too

```suggestion
name: Enrich_URL_with_urlscan.io_from_Microsoft_Teams
```

File: workflows/URLScan_with_urlscanio_from_Microsoft_Teams/help.md (hunk 2 above)
Comment (jschipp-r7) [reply workflows/URLScan_with_urlscanio_from_Microsoft_Teams/help.md:7]: Change the Teams command to this per the new convention document

File: workflows/URLScan_with_urlscanio_from_Microsoft_Teams/help.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
+# Description
+
+This workflow looks up a URL given via Slack command in urlscan.io and reports information on the URL back to Microsoft Teams.
```
Comment (sengleson-r7) [inline workflows/URLScan_with_urlscanio_from_Microsoft_Teams/help.md:3]:
```suggestion
Lookup and enrich suspicious URLs and domains with a simple Microsoft Teams message. URL and domain analysis provided by URLScan.io is returned in a Microsoft Teams message.
```

File: workflows/URLScan_with_urlscanio_from_Microsoft_Teams/help.md (hunk 3 above)
Comment (sengleson-r7) [inline workflows/URLScan_with_urlscanio_from_Microsoft_Teams/help.md:11]:
```suggestion
* **Investigate Indicators at Scale** - Manually investigating every reported phishing attempt is extremely difficult to scale. Automatic analysis of common phishing IOCs reduces the overhead associated with every reported incident.
* **Offload Repetitive Tasks** - Parsing out, analyzing, and enriching URLs manually can be repetitive and mind-numbing. Give that time back to your team so you can elevate your focus to identifying and resolving threats.
* **Shorten the Investigation Timeline** - Manage timely responses to real attacks by eliminating the investigation of false positives and spam. If the URLs are found to be malicious, pivot seamlessly to incident response.```

File: workflows/URLScan_with_urlscanio_from_Microsoft_Teams/help.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+`!urlscan https://bit.ly/39HoYMu`
+
+# Key Features
+
+* Lookup URL from Microsoft Teams
+
+# Requirements
+
+* Microsoft Teams Connection
```
Comment (sengleson-r7) [inline workflows/URLScan_with_urlscanio_from_Microsoft_Teams/help.md:15]:
```suggestion
* [Microsoft Teams](https://insightconnect.help.rapid7.com/docs/microsoft-teams)
```

File: workflows/URLScan_with_urlscanio_from_Microsoft_Teams/help.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+
+Once the workflow has been imported, each Microsoft Teams step will need the team name and channel name updated to suit your Teams environment (edit the input with the preset text of `change_me`).
+The following are the 5 Microsoft Teams steps:
+- _!urlscan Trigger_
+- _Print No URL Found_
+- _Print Failed to get Scan ID_
+- _Print Failed to get Report_
+- _Print URLScan Report_
+
+To run the workflow, send the command `!urlscan <URL>` in your configured team and channel. The workflow will post after it has completed.
```
Comment (sengleson-r7) [inline workflows/URLScan_with_urlscanio_from_Microsoft_Teams/help.md:32]:
```suggestion
# Usage

*This workflow will only trigger in the channel specified in the Microsoft Teams workflow steps.*

To run the workflow, send a message to the specified Microsoft Teams channel starting with the command `!enrich-url`.

For example:

* `!enrich-url badsite.com`
* `!enrich-url http://another.badsite.com`
```

File: workflows/URLScan_with_urlscanio_from_Microsoft_Teams/workflow.spec.yaml [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+title: "URLScan with urlscan.io from Microsoft Teams"
+description: Retrieve the full URLScan.io report of a URL using a command issued from Microsoft Teams.
+version: 1.0.0
+vendor: rapid7
+support: rapid7
+status: []
+hub_tags:
+  use_cases: [threat_detection_and_response]
+  keywords: [chat, chatops, phishing, work_from_home]
```
Comment (sengleson-r7) [inline workflows/URLScan_with_urlscanio_from_Microsoft_Teams/workflow.spec.yaml:12]:
```suggestion
  keywords: [chat, chatops, phishing, work_from_home, microsoft_teams]
```

File: workflows/URLScan_with_urlscanio_from_Microsoft_Teams/URLScan_with_Microsoft_Teams.icon [hunk 10]
```diff
@@ -0,0 +1,1728 @@
...
+              "slugVersion": "2.0.2",
+              "imageData": "https://us.cdn-assets.connect.insight.rapid7.com/icons/rapid7/microsoft_teams/2.0.2/icon.png"
+            },
+            "identifier": "new_message_received",
+            "continueOnFailure": false,
+            "isDisabled": false,
+            "parameters": {
+              "input": {
+                "channel_name": "change_me",
+                "message_content": "^!urlscan.*",
```
Comment (sengleson-r7) [inline workflows/URLScan_with_urlscanio_from_Microsoft_Teams/URLScan_with_Microsoft_Teams.icon:1307]:
```suggestion
                "message_content": "^!enrich-url.*",
```

File: workflows/URLScan_with_urlscanio_from_Microsoft_Teams/URLScan_with_Microsoft_Teams.icon [hunk 11]
```diff
@@ -0,0 +1,1728 @@
...
+        "tags": null,
+        "humanCostSeconds": 0,
+        "humanCostDisplayUnit": ""
+      }
+    ],
+    "triggers": [
+      {
+        "id": "53df9662-01db-4512-b5ee-a87ede9a63f2",
+        "identifier": "new_message_received",
+        "name": "!urlscan Trigger",
```
Comment (sengleson-r7) [inline workflows/URLScan_with_urlscanio_from_Microsoft_Teams/URLScan_with_Microsoft_Teams.icon:1565]:
```suggestion
        "name": "!enrich-url Trigger",
```

File: workflows/URLScan_with_urlscanio_from_Microsoft_Teams/help.md [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+
+# Documentation
+
+## Setup
+
+Import the workflow from the Rapid7 Extension Library and proceed through the Import Workflow wizard in InsightConnect. Import plugins, create or select connections, and rename the workflow as a part of the Import Workflow wizard as necessary.
+
+Once the workflow has been imported, each Microsoft Teams step will need the team name and channel name updated to suit your Teams environment (edit the input with the preset text of `change_me`).
+The following are the 5 Microsoft Teams steps:
+- _!urlscan Trigger_
```
Comment (sengleson-r7) [inline workflows/URLScan_with_urlscanio_from_Microsoft_Teams/help.md:26]:
```suggestion
- _!enrich-url Trigger_
```

File: workflows/URLScan_with_urlscanio_from_Microsoft_Teams/URLScan_with_Microsoft_Teams.icon [hunk 13]
```diff
@@ -0,0 +1,1728 @@
...
+              "title": "Variables",
+              "type": "object"
+            },
+            "outputJSONSchema": null,
+            "defaultImageData": "",
+            "connectionType": "CONNECTION"
+          },
+          "9dfbea14-3d78-4fbb-937b-88d4c9b277d8": {
+            "nodeId": "9dfbea14-3d78-4fbb-937b-88d4c9b277d8",
+            "name": "!urlscan Trigger",
```
Comment (sengleson-r7) [inline workflows/URLScan_with_urlscanio_from_Microsoft_Teams/URLScan_with_Microsoft_Teams.icon:1292]:
```suggestion
            "name": "!enrich-url Trigger",
```

File: workflows/URLScan_with_urlscanio_from_Microsoft_Teams/URLScan_with_Microsoft_Teams.icon [hunk 14]
```diff
@@ -0,0 +1,1728 @@
...
+    ],
+    "triggers": [
+      {
+        "id": "53df9662-01db-4512-b5ee-a87ede9a63f2",
+        "identifier": "new_message_received",
+        "name": "!urlscan Trigger",
+        "description": "",
+        "input": {
+          "channel_name": "change_me",
+          "message_content": "^!urlscan.*",
```
Comment (sengleson-r7) [inline workflows/URLScan_with_urlscanio_from_Microsoft_Teams/URLScan_with_Microsoft_Teams.icon:1569]:
```suggestion
          "message_content": "^!enrich-url.*",
```

File: workflows/URLScan_with_urlscanio_from_Microsoft_Teams/help.md (hunk 2 above)
Comment (tylerslijboom-r7) [reply workflows/URLScan_with_urlscanio_from_Microsoft_Teams/help.md:7]: I have updated the ICON file

File: workflows/Enrich_URL_with_urlscanio_from_Microsoft_Teams/Enrich_URL_with_urlscanio_from_Microsoft_Teams.icon [hunk 15]
```diff
@@ -0,0 +1,1728 @@
...
+  "kom": {
+    "komandVersion": "1.49.2-0-gafb52",
+    "komFileVersion": "2.0.0",
+    "exportedAt": "2020-04-23T16:48:11.225694111Z",
+    "workflowVersions": [
+      {
+        "name": "Enrich URL with urlscan.io from Microsoft Teams",
+        "type": "runnable",
+        "version": "",
+        "description": "Retrieve the full URLScan.io report of a URL using a command issued from Microsoft Teams.",
```
Comment (mrinehart-r7) [inline workflows/Enrich_URL_with_urlscanio_from_Microsoft_Teams/Enrich_URL_with_urlscanio_from_Microsoft_Teams.icon:11]:
```suggestion
        "description": "Retrieve the full urlscan.io report of a URL using a command issued from Microsoft Teams.",
```

File: workflows/Enrich_URL_with_urlscanio_from_Microsoft_Teams/Enrich_URL_with_urlscanio_from_Microsoft_Teams.icon [hunk 16]
```diff
@@ -0,0 +1,1728 @@
...
+              "name": "",
+              "description": "",
+              "parentNodeId": "",
+              "fromNodeId": "9dfbea14-3d78-4fbb-937b-88d4c9b277d8",
+              "toNodeId": "5ba8be26-3759-4e53-b3c8-86dfd6ecad58"
+            },
+            "265ddfe1-eb26-4ee8-909a-9fe3b963b0bf": {
+              "id": "265ddfe1-eb26-4ee8-909a-9fe3b963b0bf",
+              "name": "No Result",
+              "description": "Unable to get the Report Results from URLScan.io",
```
Comment (mrinehart-r7) [inline workflows/Enrich_URL_with_urlscanio_from_Microsoft_Teams/Enrich_URL_with_urlscanio_from_Microsoft_Teams.icon:49]:
```suggestion
              "description": "Unable to get the Report Results from urlscan.io",
```

File: workflows/Enrich_URL_with_urlscanio_from_Microsoft_Teams/Enrich_URL_with_urlscanio_from_Microsoft_Teams.icon [hunk 17]
```diff
@@ -0,0 +1,1728 @@
...
+              "name": "No Result",
+              "description": "Failed to find a URL from the user",
+              "parentNodeId": "",
+              "fromNodeId": "9108708f-fe22-4f2a-869b-7ba004878a27",
+              "toNodeId": "41587ce0-e706-4d30-9332-90f43445f90a"
+            },
+            "d13de5d6-ea32-4837-aebe-59145a20f0da": {
+              "id": "d13de5d6-ea32-4837-aebe-59145a20f0da",
+              "name": "Not Accepted",
+              "description": "URLScan.io service did not accept the URL, no scan ID",
```
Comment (mrinehart-r7) [inline workflows/Enrich_URL_with_urlscanio_from_Microsoft_Teams/Enrich_URL_with_urlscanio_from_Microsoft_Teams.icon:129]:
```suggestion
              "description": "urlscan.io service did not accept the URL, no scan ID",
```

File: workflows/Enrich_URL_with_urlscanio_from_Microsoft_Teams/Enrich_URL_with_urlscanio_from_Microsoft_Teams.icon [hunk 18]
```diff
@@ -0,0 +1,1728 @@
...
+              "slugVersion": "2.0.2",
+              "imageData": "https://us.cdn-assets.connect.insight.rapid7.com/icons/rapid7/microsoft_teams/2.0.2/icon.png"
+            },
+            "identifier": "send_html_message",
+            "continueOnFailure": false,
+            "isDisabled": false,
+            "parameters": {
+              "input": {
+                "channel_name": "change_me",
+                "message_content": "🔒URLScan.io Report for URL:🔒<br>\n<b>{{[e05bb28e-85ff-4ddb-a3f5-cef9a1676ab0].[content]}}</b><br>\nTotal HTTP Requests: {{[405042a8-112f-48f3-9ed1-9434e3c08bdc].[requests]}} <br>\nTotal Cookies: {{[405042a8-112f-48f3-9ed1-9434e3c08bdc].[cookies]}} <br>\nTotal Links: {{[405042a8-112f-48f3-9ed1-9434e3c08bdc].[links]}} <br>\nTotal JSON Globals: {{[405042a8-112f-48f3-9ed1-9434e3c08bdc].[json]}} <br>\n<b>✅ Workflow Complete</b>",
```
Comment (mrinehart-r7) [inline workflows/Enrich_URL_with_urlscanio_from_Microsoft_Teams/Enrich_URL_with_urlscanio_from_Microsoft_Teams.icon:240]:
```suggestion
                "message_content": "🔒urlscan.io Report for URL:🔒<br>\n<b>{{[e05bb28e-85ff-4ddb-a3f5-cef9a1676ab0].[content]}}</b><br>\nTotal HTTP Requests: {{[405042a8-112f-48f3-9ed1-9434e3c08bdc].[requests]}} <br>\nTotal Cookies: {{[405042a8-112f-48f3-9ed1-9434e3c08bdc].[cookies]}} <br>\nTotal Links: {{[405042a8-112f-48f3-9ed1-9434e3c08bdc].[links]}} <br>\nTotal JSON Globals: {{[405042a8-112f-48f3-9ed1-9434e3c08bdc].[json]}} <br>\n<b>✅ Workflow Complete</b>",
```

File: workflows/Enrich_URL_with_urlscanio_from_Microsoft_Teams/Enrich_URL_with_urlscanio_from_Microsoft_Teams.icon [hunk 19]
```diff
@@ -0,0 +1,1728 @@
...
+              "slugVersion": "2.0.2",
+              "imageData": "https://us.cdn-assets.connect.insight.rapid7.com/icons/rapid7/microsoft_teams/2.0.2/icon.png"
+            },
+            "identifier": "send_html_message",
+            "continueOnFailure": false,
+            "isDisabled": false,
+            "parameters": {
+              "input": {
+                "channel_name": "change_me",
+                "message_content": "🔒URLScan.io🔒<br>\nFailed to find URL from message: <br>\n{{[9dfbea14-3d78-4fbb-937b-88d4c9b277d8].[message].[body].[content]}} <br>\n❌<b>Workflow Failed</b>",
```
Comment (mrinehart-r7) [inline workflows/Enrich_URL_with_urlscanio_from_Microsoft_Teams/Enrich_URL_with_urlscanio_from_Microsoft_Teams.icon:535]:
```suggestion
                "message_content": "🔒urlscan.io🔒<br>\nFailed to find URL from message: <br>\n{{[9dfbea14-3d78-4fbb-937b-88d4c9b277d8].[message].[body].[content]}} <br>\n❌<b>Workflow Failed</b>",
```

File: workflows/Enrich_URL_with_urlscanio_from_Microsoft_Teams/Enrich_URL_with_urlscanio_from_Microsoft_Teams.icon [hunk 20]
```diff
@@ -0,0 +1,1728 @@
...
+              "slugVersion": "2.0.2",
+              "imageData": "https://us.cdn-assets.connect.insight.rapid7.com/icons/rapid7/microsoft_teams/2.0.2/icon.png"
+            },
+            "identifier": "send_message",
+            "continueOnFailure": false,
+            "isDisabled": false,
+            "parameters": {
+              "input": {
+                "channel_name": "change_me",
+                "message": "🔒URLScan.io🔒<br>\nFailed to get a Report for {{[e05bb28e-85ff-4ddb-a3f5-cef9a1676ab0].[content]}} <br>\n❌<b>Workflow Failed</b>\n",
```
Comment (mrinehart-r7) [inline workflows/Enrich_URL_with_urlscanio_from_Microsoft_Teams/Enrich_URL_with_urlscanio_from_Microsoft_Teams.icon:812]:
```suggestion
                "message": "🔒urlscan.io🔒<br>\nFailed to get a Report for {{[e05bb28e-85ff-4ddb-a3f5-cef9a1676ab0].[content]}} <br>\n❌<b>Workflow Failed</b>\n",
```

File: workflows/Enrich_URL_with_urlscanio_from_Microsoft_Teams/Enrich_URL_with_urlscanio_from_Microsoft_Teams.icon [hunk 21]
```diff
@@ -0,0 +1,1728 @@
...
+              "slugVersion": "2.0.2",
+              "imageData": "https://us.cdn-assets.connect.insight.rapid7.com/icons/rapid7/microsoft_teams/2.0.2/icon.png"
+            },
+            "identifier": "send_html_message",
+            "continueOnFailure": false,
+            "isDisabled": false,
+            "parameters": {
+              "input": {
+                "channel_name": "change_me",
+                "message_content": "🔒URLScan.io🔒<br>\nFailed to get a report for {{[e05bb28e-85ff-4ddb-a3f5-cef9a1676ab0].[content]}} <br>\n❌<b>Workflow Failed</b>",
```
Comment (mrinehart-r7) [inline workflows/Enrich_URL_with_urlscanio_from_Microsoft_Teams/Enrich_URL_with_urlscanio_from_Microsoft_Teams.icon:1137]:
```suggestion
                "message_content": "🔒urlscan.io🔒<br>\nFailed to get a report for {{[e05bb28e-85ff-4ddb-a3f5-cef9a1676ab0].[content]}} <br>\n❌<b>Workflow Failed</b>",
```

File: workflows/Enrich_URL_with_urlscanio_from_Microsoft_Teams/help.md [hunk 22]
```diff
@@ -0,0 +1,68 @@
+# Description
+
+Lookup and enrich suspicious URLs and domains with a simple Microsoft Teams message. URL and domain analysis provided by URLScan.io is returned in a Microsoft Teams message.
```
Comment (mrinehart-r7) [inline workflows/Enrich_URL_with_urlscanio_from_Microsoft_Teams/help.md:3]:
```suggestion
Lookup and enrich suspicious URLs and domains with a simple Microsoft Teams message. URL and domain analysis provided by urlscan.io is returned in a Microsoft Teams message.
```

File: workflows/Enrich_URL_with_urlscanio_from_Microsoft_Teams/help.md [hunk 23]
```diff
@@ -0,0 +1,68 @@
...
+# Key Features
+
+* **Investigate Indicators at Scale** - Manually investigating every reported phishing attempt is extremely difficult to scale. Automatic analysis of common phishing IOCs reduces the overhead associated with every reported incident.
+* **Offload Repetitive Tasks** - Parsing out, analyzing, and enriching URLs manually can be repetitive and mind-numbing. Give that time back to your team so you can elevate your focus to identifying and resolving threats.
+* **Shorten the Investigation Timeline** - Manage timely responses to real attacks by eliminating the investigation of false positives and spam. If the URLs are found to be malicious, pivot seamlessly to incident response.
+
+# Requirements
+
+* [Microsoft Teams](https://insightconnect.help.rapid7.com/docs/microsoft-teams)
+* URLScan.io API Key
```
Comment (mrinehart-r7) [inline workflows/Enrich_URL_with_urlscanio_from_Microsoft_Teams/help.md:18]:
```suggestion
* urlscan.io API Key
```

**Pull request merged**